core/chain_manager.go

```diff
@@ -616,14 +615,12 @@ func (self *ChainManager) InsertChain(chain types.Blocks) (int, error) {
 		stats      struct{ queued, processed, ignored int }
 		tstart     = time.Now()
 
-		nonceDone    = make(chan nonceResult, len(chain))
-		nonceQuit    = make(chan struct{})
 		nonceChecked = make([]bool, len(chain))
 	)
 
 	// Start the parallel nonce verifier.
-	go verifyNonces(self.pow, chain, nonceQuit, nonceDone)
-	defer close(nonceQuit)
+	nonceAbort, nonceResults := verifyBlockNonces(self.pow, chain)
+	defer close(nonceAbort)
```

Comment:
No. If we would close it, than a subsequent write would panic (writing is not allowed on a closed channel). It doesn't really matter if the channel is not closed, the GC will still clean it up. Closing is only used to signal (readers) that the channel reached its end.